File: src/wordforge/web/schemas/audit.py
"""Audit log response schemas."""
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class EditorBrief(BaseModel):
    id: int
    display_name: str


class AuditItem(BaseModel):
    id: int
    word_id: int
    field_path: str
    target_id: int | None
    op: str
    old_value: Any
    new_value: Any
    editor: EditorBrief
    created_at: datetime


class AuditListResponse(BaseModel):
    items: list[AuditItem]
    next_cursor: str | None

File: frontend/src/api/audit.ts
import { useQuery, keepPreviousData } from '@tanstack/react-query';
import client from './client';
import type { Envelope } from './types';

export type AuditEditor = {
  id: number;
  display_name: string;
};

export type AuditItem = {
  id: number;
  word_id: number;
  field_path: string;
  target_id: number | null;
  op: string;
  old_value: unknown;
  new_value: unknown;
  editor: AuditEditor;
  created_at: string;
};

export type AuditListResponse = {
  items: AuditItem[];
  next_cursor: string | null;
};

export type AuditListParams = {
  word_id?: string;
  editor_id?: string;
  since?: string;
  until?: string;
  cursor?: string;
  limit?: number;
};

export function useAuditListQuery(params: AuditListParams) {
  return useQuery<AuditListResponse>({
    queryKey: ['audit', 'list', params],
    queryFn: async () => {
      const res = await client.get<Envelope<AuditListResponse>>('/audit', {
        params,
      });
      return res.data.data;
    },
    placeholderData: keepPreviousData,
  });
}
